test/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(SRCROOT "${PROJECT_SOURCE_DIR}/test/src")
+
+include_directories("${PROJECT_SOURCE_DIR}/include")
+include_directories("${PROJECT_SOURCE_DIR}/extlibs/headers")
+include_directories("${SRCROOT}/TestUtilities")
+
+# System is always built
+SET(SYSTEM_SRC
+    "${SRCROOT}/CatchMain.cpp"
+    "${SRCROOT}/System/Vector2.cpp"
+    "${SRCROOT}/System/Vector3.cpp"
+    "${SRCROOT}/TestUtilities/SystemUtil.hpp"
+    "${SRCROOT}/TestUtilities/SystemUtil.cpp"
+)
+sfml_add_test(test-sfml-system "${SYSTEM_SRC}"  sfml-system)
+
+if(SFML_BUILD_WINDOW)
+    SET(WINDOW_SRC
+        "${SRCROOT}/CatchMain.cpp"
+        "${SRCROOT}/TestUtilities/WindowUtil.hpp"
+        "${SRCROOT}/TestUtilities/WindowUtil.cpp"
+    )
+    sfml_add_test(test-sfml-window "${WINDOW_SRC}" sfml-window)
+endif()
+
+if(SFML_BUILD_GRAPHICS)
+    SET(GRAPHICS_SRC
+        "${SRCROOT}/CatchMain.cpp"
+        "${SRCROOT}/Graphics/Rect.cpp"
+        "${SRCROOT}/TestUtilities/GraphicsUtil.hpp"
+        "${SRCROOT}/TestUtilities/GraphicsUtil.cpp"
+    )
+    sfml_add_test(test-sfml-graphics "${GRAPHICS_SRC}" sfml-graphics)
+endif()
+
+# Automatically run the tests at the end of the build
+add_custom_target(runtests ALL
+                  DEPENDS test-sfml-system test-sfml-window test-sfml-graphics
+)
+
+add_custom_command(TARGET runtests
+                   COMMENT "Run tests"
+                   POST_BUILD COMMAND ctest ARGS --output-on-failure
+)
```

Comment:
What was the rationale for automatically running tests after they're built? This is not a conventional thing to do and not something I've seen on other C++ projects. Now that SFML has 31 test cases, the runtime impact of these tests is starting to approach the time it takes to incrementally build a few C++ files. This problem is only prone to get worse over time as we improve test coverage and the test suite takes longer and longer to run. Our CI pipeline is so rigorous that if a contributor forgets to build or run the tests, that problem is immediately caught upon submission of their PR so I'm not particularly worried about such situations.

Reply:
https://github.com/SFML/SFML/issues/2429

I filed an issue to remove this.